shared/security/src/main/java/io/pravega/shared/security/token/JsonWebToken.java

```diff
@@ -107,6 +104,19 @@ public JsonWebToken(@NonNull String subject, @NonNull String audience, @NonNull
         this.permissionsByResource = resourcePermissionClaims;
     }
 
+    public JsonWebToken(@NonNull String subject, @NonNull String audience, @NonNull byte[] signingKey,
+                        Date expiry, Map<String, Object> resourcePermissionClaims) {
+        this.subject = subject;
+        this.audience = audience;
+        this.signingKey = signingKey.clone();
+        if (expiry != null) {
+            this.expirationTime = Date.from(expiry.toInstant());
+        } else {
+            this.expirationTime = null;
+        }
+        this.permissionsByResource = resourcePermissionClaims;
```

Comment:
Should this have a defensive copy?

Reply:
The class is used internally. I had to make a defensive copy of expiration time and bytes to satisfy Spotbugs.